src/main/java/org/elasticsearch/indices/IndicesService.java

```diff
@@ -471,12 +471,14 @@ public void deleteIndexStore(String reason, IndexMetaData metaData) throws IOExc
         if (nodeEnv.hasNodeFile()) {
             synchronized (this) {
                 String indexName = metaData.index();
-                if (indices.containsKey(metaData.index())) {
-                    String localUUid = indices.get(metaData.index()).v1().indexUUID();
-                    throw new ElasticsearchIllegalStateException("Can't delete index store for [" + metaData.getIndex() + "] - it's still part of the indices service [" + localUUid+ "] [" + metaData.getUUID() + "]");
+                if (indices.containsKey(indexName)) {
+                    String localUUid = indices.get(indexName).v1().indexUUID();
+                    throw new ElasticsearchIllegalStateException("Can't delete index store for [" + indexName + "] - it's still part of the indices service [" + localUUid+ "] [" + metaData.getUUID() + "]");
                 }
                 ClusterState clusterState = clusterService.state();
-                if (clusterState.metaData().hasIndex(indexName)) {
+                if (clusterState.metaData().hasIndex(indexName) && (clusterState.nodes().localNode().masterNode() == true)) {
```

Comment:
in this case isn't the exception wrong? I mean shouldn't we just skip it in that case?